### PR TITLE
Fix some deprecation warnings

### LINKED ISSUE #863: Fix DeprecationWarnings
Some example warnings from a failed test run:

```
conftest.py:30
  /__w/MSS/MSS/conftest.py:30: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

mslib/msui/_tests/test_mscolab.py: 18 warnings
mslib/msui/_tests/test_mscolab_admin_window.py: 7 warnings
mslib/msui/_tests/test_mscolab_project.py: 6 warnings
mslib/msui/_tests/test_mscolab_version_history.py: 6 warnings
  /opt/conda/envs/mssenv/lib/python3.9/site-packages/requests/auth.py:39: DeprecationWarning: Non-string usernames will no longer be supported in Requests 3.0.0. Please convert the object you've passed in (None) to a string or bytes object in the near future to avoid problems.
    warnings.warn(

mslib/msui/_tests/test_mscolab.py: 18 warnings
mslib/msui/_tests/test_mscolab_admin_window.py: 7 warnings
mslib/msui/_tests/test_mscolab_project.py: 6 warnings
mslib/msui/_tests/test_mscolab_version_history.py: 6 warnings
  /opt/conda/envs/mssenv/lib/python3.9/site-packages/requests/auth.py:49: DeprecationWarning: Non-string passwords will no longer be supported in Requests 3.0.0. Please convert the object you've passed in (<class 'NoneType'>) to a string or bytes object in the near future to avoid problems.
    warnings.warn(

mslib/msui/_tests/test_mscolab.py: 12 warnings
mslib/msui/_tests/test_mscolab_version_history.py: 84 warnings
mslib/msui/_tests/test_mss_pyui.py: 12 warnings
mslib/msui/_tests/test_tableview.py: 1149 warnings
  /__w/MSS/MSS/mslib/msui/flighttrack.py:196: DeprecationWarning: an integer is required (got type ItemFlags).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.
    return QtCore.Qt.ItemFlags(

mslib/msui/_tests/test_mscolab.py: 18 warnings
mslib/msui/_tests/test_mscolab_version_history.py: 42 warnings
mslib/msui/_tests/test_mss_pyui.py: 18 warnings
mslib/msui/_tests/test_tableview.py: 1577 warnings
  /__w/MSS/MSS/mslib/msui/flighttrack.py:201: DeprecationWarning: an integer is required (got type ItemFlags).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.
    return QtCore.Qt.ItemFlags(

mslib/msui/_tests/test_mscolab_project.py: 14 warnings
  /__w/MSS/MSS/mslib/msui/mscolab_project.py:486: DeprecationWarning: an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.
    text_browser.setFixedHeight(

mslib/msui/_tests/test_mscolab_project.py::Test_MscolabProject::test_edit_message
  /__w/MSS/MSS/mslib/msui/mscolab_project.py:647: DeprecationWarning: an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.
    self.messageBox.setFixedHeight(

mslib/msui/_tests/test_topview.py::Test_MSSTopViewWindow::test_move_point
mslib/msui/_tests/test_topview.py::Test_MSSTopViewWindow::test_move_point
  /__w/MSS/MSS/mslib/msui/topview.py:354: DeprecationWarning: In future, it will be an error for 'np.bool_' scalars to be interpreted as an index
    self.btRoundtrip.setEnabled(self.is_roundtrip_possible())

mslib/msui/_tests/test_wms_control.py: 18 warnings
  /opt/conda/envs/mssenv/lib/python3.9/site-packages/isodate/duration.py:183: DeprecationWarning: an integer is required (got type decimal.Decimal).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.
    newdt = other.replace(year=newyear, month=newmonth, day=newday)

mslib/mswms/_tests/test_mss_plot_driver.py::Test_VSec::test_VS_RelativeHumdityStyle_01
  /__w/MSS/MSS/mslib/mswms/mpl_vsec_styles.py:543: UserWarning: No contour levels were found within the data range.
    ax.contour(self.lat_inds.repeat(numlevel).reshape((numpoints, numlevel)).transpose(),

mslib/mswms/_tests/test_mss_plot_driver.py::Test_VSec::test_VS_HorizontalVelocityStyle_01
mslib/mswms/_tests/test_wms.py::Test_WMS::test_produce_vsec_plot
mslib/mswms/_tests/test_wms.py::Test_WMS::test_produce_vsec_service_exception
  /__w/MSS/MSS/mslib/mswms/mpl_vsec_styles.py:856: UserWarning: No contour levels were found within the data range.
    cs_v2 = ax.contour(self.lat_inds.repeat(numlevel).reshape((numpoints, numlevel)).transpose(),

mslib/mswms/_tests/test_mss_plot_driver.py::Test_VSec::test_VS_EMACEyja_Style_01
  /__w/MSS/MSS/mslib/mswms/mpl_vsec_styles.py:1376: UserWarning: Log scale: values of z <= 0 have been masked
    cs = ax.contourf(self.lat_inds.repeat(numlevel).reshape((numpoints, numlevel)).transpose(),

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_meter_crs_codes[EPSG:3031]
  /opt/conda/envs/mssenv/lib/python3.9/site-packages/matplotlib/contour.py:1430: UserWarning: Warning: converting a masked element to nan.
    self.zmax = float(z.max())

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_meter_crs_codes[EPSG:3031]
  /opt/conda/envs/mssenv/lib/python3.9/site-packages/matplotlib/contour.py:1431: UserWarning: Warning: converting a masked element to nan.
    self.zmin = float(z.min())

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_meter_crs_codes[EPSG:3031]
  /__w/MSS/MSS/mslib/mswms/mpl_hsec_styles.py:211: UserWarning: No contour levels were found within the data range.
    cs = bm.contour(lonmesh, latmesh, mslp,

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_meter_crs_codes[EPSG:3031]
  /__w/MSS/MSS/mslib/mswms/mpl_hsec_styles.py:214: UserWarning: No contour levels were found within the data range.
    cs = bm.contour(lonmesh, latmesh, mslp,

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_repeated_locations
mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_HS_TemperatureStyle_ML_01
  /__w/MSS/MSS/mslib/mswms/mpl_hsec_styles.py:435: UserWarning: No contour levels were found within the data range.
    cs = bm.contour(lonmesh, latmesh, tempC,

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_HS_SEAStyle_01
  /__w/MSS/MSS/mslib/mswms/mpl_hsec_styles.py:309: UserWarning: No contour levels were found within the data range.
    cs3 = bm.contour(lonmesh, latmesh, sea,

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_HS_GenericStyle_other
  /__w/MSS/MSS/mslib/mswms/mpl_hsec_styles.py:482: UserWarning: No contour levels were found within the data range.
    cs_pv = ax.contour(lonmesh, latmesh, self.data[cont_data], cont_levels,

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_HS_RelativeHumidityStyle_PL_01
  /__w/MSS/MSS/mslib/mswms/mpl_hsec_styles.py:816: UserWarning: No contour levels were found within the data range.
    cs = bm.contour(lonmesh, latmesh, rh,

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_HS_DivStyle_PL_01
mslib/mswms/_tests/test_wms.py::Test_WMS::test_produce_hsec_plot
mslib/mswms/_tests/test_wms.py::Test_WMS::test_produce_hsec_service_exception
mslib/mswms/_tests/test_wms.py::Test_WMS::test_application_request
mslib/mswms/_tests/test_wms.py::Test_WMS::test_application_request_lowercase
  /__w/MSS/MSS/mslib/mswms/mpl_hsec_styles.py:1053: UserWarning: No contour levels were found within the data range.
    cs = bm.contour(lonmesh, latmesh, d,

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_HS_DivStyle_PL_01
mslib/mswms/_tests/test_wms.py::Test_WMS::test_produce_hsec_plot
mslib/mswms/_tests/test_wms.py::Test_WMS::test_produce_hsec_service_exception
mslib/mswms/_tests/test_wms.py::Test_WMS::test_application_request
mslib/mswms/_tests/test_wms.py::Test_WMS::test_application_request_lowercase
  /__w/MSS/MSS/mslib/mswms/mpl_hsec_styles.py:1056: UserWarning: No contour levels were found within the data range.
    cs = bm.contour(lonmesh, latmesh, d,

mslib/mswms/_tests/test_mss_plot_driver.py::Test_HSec::test_HS_EMAC_TracerStyle_ML_01
  /__w/MSS/MSS/mslib/mswms/mpl_hsec_styles.py:1118: UserWarning: No contour levels were found within the data range.
    ac = bm.contour(lonmesh, latmesh, tracer,

```

### PR DESCRIPTION
**Pull request by joernu76**
See issue #863

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -24,10 +24,9 @@
     See the License for the specific language governing permissions and
     limitations under the License.
 """
-from __future__ import print_function
 
-
-import imp
+import importlib
+import importlib.machinery
 import os
```
Comment (ReimarBauer) [inline conftest.py:30]: nice

**Review by ReimarBauer — COMMENTED** [review]
please add in a comment the deprecation fixed

**Comment by joernu76** [comment]
Updated commit message with list of fixed issues. Lots of stuff left, but probably nothing that will fail our code in the near future.

**Comment by joernu76** [comment]
> please add in a comment the deprecation fixed

Done.

**Review by ReimarBauer — APPROVED** [review]
Thx

**Pull request merged**